src/process.rs

```diff
@@ -705,6 +705,13 @@ impl<T> PartialEq for ProcessRef<T> {
     }
 }
 
+// Implement Hash explicitly to match the behavior of PartialEq
+impl<T> std::hash::Hash for ProcessRef<T> {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.process.hash(state);
```

Comment:
I am not sure if hashing `process` is correct since `Hash` should match the behavior of `PartialEq` but `PartialEq` only compares `process.id` and not `process.node_id`.

Reply:
PartialEq is wrong and should compare both, id and node_id.

Reply:
Ok I'll fix that as well

Reply:
Fixed!